kobo/apps/subsequences/tests/test_submission_extras_api_post.py

```diff
@@ -88,6 +95,115 @@ def test_translation_revisions_stored_properly(self):
+        """
+
+        # Enable transcripts and translations for the example question
+        self.set_asset_advanced_features(
+            {
+                'transcript': {'values': ['q1']},
+                'translation': {
+                    'values': ['q1'],
+                    'languages': ['tx1', 'tx2'],
+                },
+            }
+        )
+        resp = self.client.get(self.asset_url)
+        assert resp.status_code == 200
+        schema = resp.json()['advanced_submission_schema']
+
+        # Submit transcript and translation as the owner
+        original_transcript = 'they said hello'
+        original_translation = 'T H E Y   S A I D   H E L L O'
+        package = {
+            'submission': 'abc123-def456',
+            'q1': {
+                'transcript': {
+                    'value': original_transcript,
+                    'languageCode': 'en',
+                },
+                'translation': {
+                    'tx1': {
+                        'value': original_translation,
+                        'languageCode': 'xx',
+                    }
+                },
+            },
+        }
+        resp = self.client.post(schema['url'], package, format='json')
+        assert resp.status_code == 200
+        q1_transx = resp.json()['q1']
+        assert q1_transx['transcript']['value'] == original_transcript
+        assert q1_transx['translation']['tx1']['value'] == original_translation
+
+        # Become a user with no access to the project
+        other_user = User.objects.create(username='ethan')
+        self.client.force_login(other_user)
+        modified = deepcopy(package)
+        modified_transcript = 'they said goodbye'
+        modified_translation = 'T H E Y   S A I D   G O O D B Y E'
+        modified['q1']['transcript']['value'] = modified_transcript
+        modified['q1']['translation']['tx1']['value'] = modified_translation
+
+        # Attempt to change transcript should be rejected with no permissions
+        # assigned
+        resp = self.client.post(schema['url'], modified, format='json')
+        assert resp.status_code == 404
+
+        # …and should be rejected with any of these insufficient permissions
+        # assigned
+        self.asset.assign_perm(other_user, PERM_ADD_SUBMISSIONS)
+        resp = self.client.post(schema['url'], modified, format='json')
+        assert resp.status_code == 404
+
+        self.asset.assign_perm(other_user, PERM_VIEW_ASSET)
+        resp = self.client.post(schema['url'], modified, format='json')
+        assert resp.status_code == 404
+
+        self.asset.assign_perm(other_user, PERM_CHANGE_ASSET)
+        resp = self.client.post(schema['url'], modified, format='json')
+        assert resp.status_code == 404
+
+        self.asset.assign_perm(other_user, PERM_VIEW_SUBMISSIONS)
+        resp = self.client.post(schema['url'], modified, format='json')
+        assert resp.status_code == 403
```

Comment:
Not related to this PR but I wondering why `PERM_VIEW_ASSET`, `PERM_CHANGE_ASSET` return a 404 while `PERM_VIEW_SUBMISSIONS` returns a 403. I was assuming that as soon as user had `PERM_VIEW_ASSET`, users would receive a 403 when they don't have required permissions on nested objects.

Reply:
Good question; I'd assume it's just the usual DRF behavior of:
* Did you send a GET for a specific object that you are not allowed to see? We're not telling you whether or not it exists, so 404
* Did you try to take a disallowed action (POST, PATCH) on an object that you *are* allowed to see? → 403

I didn't expect nested objects to behave differently, but we can look into it more if you think something doesn't make sense

Reply:
Right! that's what I'm expecting too. 

I missed [an important line](https://github.com/kobotoolbox/kpi/blob/require-change-not-add-for-transcripts-translations/kpi/permissions.py#L343)! 

I was thinking the model was `Asset`.  Everything is good.